### PR TITLE
fix/751 OTC Swap Flow

### PR DESCRIPTION
**Pull request by coltraneyan**
- addresses all issues from feedback in https://github.com/airswap/airswap-web/pull/755 
- uses redux to handle searchAmount state with correct cleanup
- adds AvailableSwapsWidget to OrderDetailWidget as well for smoother transitions

* all server querying is the same, however, "Aggregate Error" can happen when re-querying. Not sure if this will be fixed by Mitsi and Whatdays current work etc. but lets please tackle this in another PR.

### CODE REVIEW

File: src/components/OrderDetailWidget/OrderDetailWidget.tsx [hunk 1]
```diff
@@ -119,9 +131,45 @@ const OrderDetailWidget: FC<OrderDetailWidgetProps> = ({ order }) => {
...
+
+  useEffect(() => {
+    if (!indexerUrls && library) {
+      dispatch(fetchIndexerUrls({ provider: library }));
+    }
+  }, [dispatch, indexerUrls, library]);
+
+  useEffect(() => {
+    if (indexerUrls) {
+      dispatch(
```
Comment (piersss) [inline src/components/OrderDetailWidget/OrderDetailWidget.tsx:144]: This is called 2 times when arriving from SwapWidget. You should probably only have indexers as dependency.

File: src/components/OrderDetailWidget/OrderDetailWidget.tsx [hunk 2]
```diff
@@ -119,9 +131,45 @@ const OrderDetailWidget: FC<OrderDetailWidgetProps> = ({ order }) => {
   }, [order]);
 
   const [showFeeInfo, toggleShowFeeInfo] = useToggle(false);
+  const [showViewAllQuotes, toggleShowViewAllQuotes] = useToggle(false);
+
+  useEffect(() => {
+    if (!indexerUrls && library) {
+      dispatch(fetchIndexerUrls({ provider: library }));
+    }
+  }, [dispatch, indexerUrls, library]);
```
Comment (piersss) [inline src/components/OrderDetailWidget/OrderDetailWidget.tsx:140]:
```suggestion
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [indexerUrls, library]);
```

File: src/components/OrderDetailWidget/OrderDetailWidget.tsx [hunk 3]
```diff
@@ -119,9 +131,45 @@ const OrderDetailWidget: FC<OrderDetailWidgetProps> = ({ order }) => {
...
+            offset: 0,
+            limit: 100,
+          },
+        })
+      );
+    }
+  }, [dispatch, indexerUrls, senderToken?.address, signerToken?.address]);
+
+  // swap flow handlers
+  const userIsFromSwapFlow = !!location.state?.fromSwapFlow;
```
Comment (piersss) [inline src/components/OrderDetailWidget/OrderDetailWidget.tsx:158]: This should be higher with the variable declarations. Not between useEffect's.

File: src/components/OrderDetailWidget/OrderDetailWidget.tsx [hunk 4]
```diff
@@ -119,9 +131,45 @@ const OrderDetailWidget: FC<OrderDetailWidgetProps> = ({ order }) => {
...
+
+  const backToViewAllQuotes = () => {
```
Comment (piersss) [inline src/components/OrderDetailWidget/OrderDetailWidget.tsx:160]:
```suggestion
  const backToSwapPage = () => {
```

File: src/components/Overlay/Overlay.tsx [hunk 5]
```diff
@@ -31,6 +31,7 @@ export type OverlayProps = {
    * Hide or show the component
    */
   isHidden?: boolean;
+  animate?: boolean;
```
Comment (piersss) [inline src/components/Overlay/Overlay.tsx:34]:
```suggestion
  shouldAnimate?: boolean;
```

**Review by piersss — CHANGES_REQUESTED** [review]
This works a lot better. Adding AvailableOrdersWidget to OrderDetailWidget is a good move.

Note: the translation is changed from `orders.availableSwaps` to `orders.availableOrders`.

Small bug when routing directly to swap from order detail widget:

https://github.com/airswap/airswap-web/assets/86911296/53a403d5-8fcb-4345-9936-f898cb342c57

**Review by piersss — CHANGES_REQUESTED** [review]
Why don't you use tradeTerms.baseAmount instead of indexer.currentSearchAmount?

Please rebase your branch to develop / merge develop into your branch.

File: src/components/OrderDetailWidget/OrderDetailWidget.tsx [hunk 6]
```diff
@@ -118,10 +130,47 @@ const OrderDetailWidget: FC<OrderDetailWidgetProps> = ({ order }) => {
...
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [indexerUrls, library]);
+
+  useEffect(() => {
+    if (indexerUrls && senderToken?.address && signerToken?.address) {
+      dispatch(
+        getFilteredOrders({
+          filter: {
+            senderTokens: [senderToken.address],
```
Comment (piersss) [inline src/components/OrderDetailWidget/OrderDetailWidget.tsx:150]: To get orders with similar amounts we should add `senderMinAmount` and `senderMaxAmount` -25% + 25% respectively.

Comment (coltraneyan) [reply src/components/OrderDetailWidget/OrderDetailWidget.tsx:150]: i had this implemented before and we decided to take it out. if we want to do this, lets move it to another PR.

**Comment by coltraneyan** [comment]
> Why don't you use tradeTerms.baseAmount instead of indexer.currentSearchAmount?
> 
> Please rebase your branch to develop / merge develop into your branch.

i'm pretty sure the components cleanup functions will clear the trade terms. current search amount is just essentially a pinned value we need for a very specific point in time, which i think is best to have its own state rather than piggyback on another.

**Review by piersss — APPROVED** [review]
LGTM

**Pull request merged**